library/src/methods/brand/brand.ts

```diff
@@ -84,9 +86,46 @@ type BrandName = string | number | symbol;
  * Brand type.
  */
 export type Brand<TBrandName extends BrandName> = {
-  [symbol]: TBrandName;
+  [BrandSymbol]: {
+    [N in TBrandName]: N;
+  };
 };
 
+type UnionToIntersection<T> = (
+  T extends never ? never : (arg: T) => never
+) extends (arg: infer U) => never
+  ? U
+  : never;
+
+/**
+ * Extracts all brands from a branded type.
+ *
+ * @example
+ * Brands<string & Brand<'foo'> & Brand<'bar'>> = Brand<'foo'> & Brand<'bar'>
+ */
+export type Brands<BrandedType> = BrandedType extends Brand<any>
+  ? UnionToIntersection<
+      {
+        [N in keyof BrandedType[BrandSymbol]]: N extends
+          | string
+          | symbol
+          | number
+          ? Brand<N>
+          : never;
+      }[keyof BrandedType[BrandSymbol]]
+    >
+  : never;
+
+/**
+ * Removes all brands from a branded type or returns the type if it is not branded.
+ * @example
+ * Unbranded<string & Brand<'foo'> & Brand<'bar'>> = string
+ */
+export type Unbranded<BrandedType> = BrandedType extends infer T &
```

Comment:
If the newly added types are only needed in the tests, they should not be in the `brand.ts` file, but directly under `brand.test.ts`. Or alternatively under `types.ts` in the `brand` folder if there is a practical use case for it outside the library code.